code/obj/mining.dm

```diff
@@ -701,161 +702,82 @@
-				return
-
-			var/html = ""
-			for(var/X in mining_controls.mining_encounters_selectable)
-				var/datum/mining_encounter/E = mining_controls.mining_encounters_selectable[X]
-				if(istype(E))
-					html += "<A href='?src=\ref[src];activate_selectable=[X]'>[E.name]</A><BR>"
-
-			html += "<BR><A href='?src=\ref[src];back=1'>Back</A><BR>"
-			usr.Browse(html, "window=computer;size=300x400")
-			onclose(usr, "computer")
-			return
-
-		else if (href_list["activate_selectable"])
-			if (src.uses_global_controls && !istype(mining_controls.magnet_area))
-				boutput(usr, "Uh oh, something's gotten really fucked up with the magnet system. Please report this to a coder!")
-				return
-
-			if (src.check_for_unacceptable_content())
-				src.visible_message("<b>[src.name]</b> states, \"Safety lock engaged. Please remove all personnel and vehicles from the magnet area.\"")
-			else
-				SPAWN(0)
-					if (src) src.pull_new_source(href_list["activate_selectable"])
-
-		else if (href_list["activate_magnet"])
-			if (src.uses_global_controls && !istype(mining_controls.magnet_area))
-				boutput(usr, "Uh oh, something's gotten really fucked up with the magnet system. Please report this to a coder!")
-				return
-
-			if (src.check_for_unacceptable_content())
-				src.visible_message("<b>[src.name]</b> states, \"Safety lock engaged. Please remove all personnel and vehicles from the magnet area.\"")
-			else
-				SPAWN(0)
-					if (src) src.pull_new_source()
+					src.last_use_attempt = TIME + 10
+					SPAWN(0)
+						if (src)
+							src.pull_new_source(params["encounter_id"])
+					. = TRUE
+			if ("activatemagnet")
+				if (magnetNotReady)
+					return
+				if (src.uses_global_controls && !istype(mining_controls.magnet_area))
+					boutput(usr, "Uh oh, something's gotten really fucked up with the magnet system. Please report this to a coder!")
+					return
 
-		else if (href_list["override_cooldown"])
-			if (!ishuman(usr))
-				boutput(usr, "<span class='alert'>AI and robotic personnel may not access the override.</span>")
-			else
-				var/mob/living/carbon/human/H = usr
-				if(!src.allowed(H))
-					boutput(usr, "<span class='alert'>Access denied. Please contact the Chief Engineer or Captain to access the override.</span>")
+				if (src.check_for_unacceptable_content())
+					src.visible_message("<b>[src.name]</b> states, \"Safety lock engaged. Please remove all personnel and vehicles from the magnet area.\"")
 				else
-					src.cooldown_override = !src.cooldown_override
-
-		else if (href_list["auto_mode"])
-			src.automatic_mode = !src.automatic_mode
-
-		else if (href_list["geo_scan"])
-			var/MC = src.get_magnetic_center()
-			if (!MC)
-				boutput(usr, "Error. Magnet is not magnetized.")
-				src.updateUsrDialog()
-				return
+					src.last_use_attempt = TIME + 10 // This is to prevent href exploits or autoclickers from pulling multiple times simultaneously
+					SPAWN(0)
+						if (src)
+							src.pull_new_source()
```

Comment:
Sanity check on why we need to bump this out?

Reply:
I need a bit more clarification

Reply:
why spawn